lib/services/user_auth.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:hotel_manager/services/user_service.dart';
+import 'package:hotel_manager/utils/Roles.dart';
+import 'package:hotel_manager/models/user_model.dart' as model;
+
+class UserAuth{
+  final UserService _userService = UserService();
+
+
+  Future<UserCredential?> signIn(String login, String password) async {
+    try {
+      UserCredential user = await FirebaseAuth.instance.signInWithEmailAndPassword(
+          email: login,
+          password: password);
+      return user;
+    }
+    catch (e) {
+      // Handle the error and return null or throw an exception if needed.
```

Comment:
rzuc wyjatkami a nie tylko print

Reply:
Throw nie zwraca nulla, który pozwala na łatwiejszą implementację w zewnętrznym kodzie, zamiast cały kod potem pakować to do try catchy